pilot/pkg/xds/endpoints/endpoint_builder.go

```diff
@@ -788,3 +822,59 @@ func getSubSetLabels(dr *v1alpha3.DestinationRule, subsetName string) labels.Ins
 
 	return nil
 }
+
+// For services that have a waypoint, we want to send to the waypoints rather than the service endpoints.
+// Lookup the
+func (b *EndpointBuilder) findServiceWaypoint(endpointIndex *model.EndpointIndex) ([]*model.IstioEndpoint, bool) {
+	if b.nodeType != model.Router {
+		// Currently only ingress will call waypoints
+		return nil, false
+	}
+	// ...and they must explicitly opt in
+	if b.service.Attributes.Labels["istio.io/ingress-use-waypoint"] != "true" {
```

Comment:
nit: This should probably (like all the other `istio.io` labels AFAIK) be a centralized constant?

Reply:
https://github.com/istio/api/pull/3307 does this fwiw

Reply:
Or rather -- under `constants` is not even the correct usage, it needs to go into istio/api really. So my code is wrong, but its wrong for multiple reasons not just one :-)